study-group/Google-Toast-Implementations.md

```diff
@@ -0,0 +1,101 @@
+
+### Design & Sample Code
+To use `<paper-toast>`, use something like this in HTML:
+
+```html
+<paper-toast text="Hello world!" opened></paper-toast>
+```
+
+This will display a notification with the text "Hello World!" that will stay open for 3 seconds (default).
+To change the default length of time it displays, set the `duration` attribute.
+
+```html
+<paper-toast text="This message closes in 5 seconds!" duration="5000" opened></paper-toast>
+```
+
+To control when the toast opens and closes, use the `.open()` and `.close()` methods from JavaScript - or use buttons to call those methods.
+For the toast to persist until `.close()` is called, set `duration` to 0.
+
+```html
+<button onclick="toast1.open()">Persistent toast</button>
+
+<paper-toast id="toast1" duration="0" text="This toast sticks around!">
+  <button onclick="toast1.toggle()">Close it!</button>
+</paper-toast>
+```
+
+To put a `<paper-toast>` into a specific container, set the `fitInto` attribute to the id of that container.
+
+```html
+<div id="container0">
+...
+</div>
+
+<paper-toast id="toast2" fitInto="container0" text="This fits perfectly!"></paper-toast>
+```
+
+Finally, to style the toast set the CSS properties `--paper-toast-background-color` and `--paper-toast-color` on the `id` of the toast in the stylesheet.
+
+### Notable Features & Details
+The Polymer `<paper-toast>` library comes built with a lot attributes for granular control, and I explore some of these attributes and design choices in this section.
+
+#### Permits Custom HTML
+`<paper-toast>` allso the developer to customize the contents of the toast with custom HTML passed in as children.
+
+#### Intuitive Positioning
+`horizontalAlign` and `verticalAlign` provide attributes to position the toast intuitively.
+
+#### Dismissal
+The `<paper-toast>` holds onto information about how it was dismissed the last time it was shown, in attributes called `canceled` and `closingReason`.
+
+#### Focus
+A few of the minute controls concern focus. The `noAutoFocus` attribute, which defaults to true (preventing autofocus), allows the user to decide if the `<paper-toast>` or its children will disable autofocus. 
+The `restoreFocusOnClear` attribute restores page's focus once the toast is cleared from the screen.
+
+#### Notes
+- There are many granular controls, concerning things like parent elements and sizing and alignment inheritance.
+- The component comes with events, for when the toast opens and the alignment changes (*not for when it closes though*).
+- Only one `<paper-toast>` will be visible on the screen at a time. If you trigger a second it will replace the first.
+- **Warning**: `<paper-toast>` is affected by the stacking context of its container, so to guarantee it shows up on the top layer it must be placed at the top level (`<body>`) element.
+    - How to properly handle top layer is an ongoing debate (see [here](https://github.com/whatwg/html/issues/897#issuecomment-198512778))
+
+### Key Takeaways
```

Comment:
This section might be unnecessary, if we're looking to make things more scalable by decreasing per-toast work.

Reply:
If you're referring to `Key Takeaways` I agree, I'm going to move that content to the findings heading for now, and integrate it with my findings for other libraries as I finish them.

Reply:
Scratch that, I just integrated it into notes, will make each library go quicker and the information was easily distillable in the notes.